src/main/java/com/example/authservice/messaging/CustomMessage.java

```diff
@@ -0,0 +1,20 @@
+package com.example.authservice.messaging;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import java.util.Date;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+public class CustomMessage {
+
+    private String messageId;
+    private String message;
+    private Date messageDate;
```

Comment:
```suggestion
    private String message;
```

We have been requested to publish a message with the new user's _"username"_ and therefore the other fields are not necessary unless UserService have specifically requested this.

Reply:
Does rabbitTemplate require a unique id?

Reply:
I'm not sure about the id, I guess we don't have any functionality that uses it? Are we going to need it for anything in the final project?

Reply:
The only thing that they requested from us was the email address. You can test and if the exchange still works then we can remove it.